### ISSUE TITLE
Use a grammar for command line options

### ISSUE THREAD

**Issue by JJ** [Raku/problem-solving#187]
The command line options described in [S19](https://design.raku.org/S19.html#Command_Line_Elements) are not fully implemented. Specifically, "delimited" options do not seem to be there, and colons (:) don't work as option prefix either. 

This probably needs a grammar. An attempt to provide one was created in rakudo/rakudo#324, but it's drifted apart from the codebase, and a consensus on how to actually to it (or if it's really needed) was not reached. Maybe we should reopen that debate, and re-use what parts of the code there are still applicable.

**Comment by AlexDaniel** [comment]
I don't think there's any debate, we do indeed need a better parser for command line options.

**Comment by niner** [comment]
I don't think this needs discussion as much as it needs someone to push forward with a working implementation. Once we have an implementation that's demonstrably better, we can discuss replacing what we have in rakudo with the new implementation. However, I don't expect that discussion to be long because if it's better, why wouldn't we use it?

So really, people dissatisfied with our command line parser just need to start writing code. This is perfectly possible as a module.
